lib/directory-tree.js

```diff
@@ -37,6 +37,12 @@ function directoryTree (path, options, onEachFile) {
 			options.extensions.indexOf(ext) === -1)
 			return null;
 
+		
+		// Skip file if it matches the exclude regex
```

Comment:
I made the changes and also made it a little more condensed. I think the PR is ready now.